packages/core-data/src/queried-data/reducer.js

```diff
@@ -73,6 +73,23 @@ export function getMergedItemIds( itemIds, nextItemIds, page, perPage ) {
 	return mergedItemIds;
 }
 
+/**
+ * Helper function to filter out entities with certain IDs.
+ * Entities are keyed by their ID.
+ *
+ * @param {Object} entities Entity objects, keyed by entity ID.
+ * @param {Array}  ids      Entity IDs to filter out.
+ *
+ * @return {Object} Filtered entities.
+ */
+function removeEntitiesById( entities, ids ) {
+	return Object.fromEntries(
+		Object.entries( entities ).filter(
+			( [ id ] ) => ! ids.some( ( itemId ) => +itemId === +id )
```

Comment:
@tyxla, do you know if `omit` also casts values internally?

Not every entity ID will be an `int`. e.g., Templates and Template parts use string for IDs.

```
wp.data.select('core').getEntityRecords( 'postType', 'wp_template' );
```

Reply:
Yes, it does. It calls `toKey` on each value:

```
    function toKey(value) {
      if (typeof value == 'string' || isSymbol(value)) {
        return value;
      }
      var result = (value + '');
      return (result == '0' && (1 / value) == -INFINITY) ? '-0' : result;
    }
```

Thanks, @Mamaduka, I'll have to address your feedback and take strings into consideration there.

Reply:
Hey, @Mamaduka in https://github.com/WordPress/gutenberg/pull/46674/commits/26425eb433849692a37e956e1902147f9559f7c2 we're now supporting both ID and string IDs. I don't think we need to overcomplicate it to support other NaN values or symbols.

Please, take another look when you have a chance!

Reply:
I think that's a good idea @Mamaduka, thanks! Test that covers support of string IDs added in 8964b9c43e4c289af803ed137af456ba81e24d01